### ISSUE TITLE
ddl.dic: allowed _type.purpose values for data items containing the _name.linked_item_id data item

### ISSUE THREAD

**Issue by vaitkus** [COMCIFS/cif_core#73]
The ``_name.linked_item_id`` data item defines a relation between two data items (usually as a primary/foreign key or value/su pair). In the ``cif_core`` dictionary data items having the  ``_name.linked_item_id`` tag are usually assigned either **SU** or **Link** ``_type.purpose`` value; however, in some cases the **Encode** purpose seems to be used interchangeably with **Link**. Are there any formal restrictions on the ``_type.purpose`` values of linking data items?

**Comment by jamesrhester** [comment]
There are none, but I think we can institute a formal rule that such data names cannot Encode something that they simply link to.

**Comment by vaitkus** [comment]
Acknowledged. For me it was more of a question how linking data items with data types **not** from the [**SU**, **Link**] set should be handled (if they are allowed at all).

**Link** implies that the values in the both data items must match (one being the subset of the other); **SU** holds no such requirement.  From the definition of the ``_name.linked_item_id`` data item it is not entirely clear how the ``_type.purpose`` value affects the interpretation of the ``_name.linked_item_id``:
```
Dataname of an equivalent item which has a common set of values, or,
in the definition of a type SU item is the name of the associated Measurement
item to which the standard uncertainty applies.
```

It could be implied that only the SU data items are handled in a special way (as having related SU values) and data items with other ``_type.purpose`` values should all be handled in the same way (as having a common set of values). Is this interpretation correct?

**Comment by jamesrhester** [comment]
I'm not sure I understand your question. If `_name.linked_item_id` is present in a definition, either `_type.purpose` is **Link**, or it is **SU**.  No other `_type.purposes` should be provided. Does that clarify things?

**Comment by vaitkus** [comment]
Yes, totally. The fact that the some ``_type.purposes`` values become invalid based on the ``_name.linked_item_id`` presence, however, does send a signal that the system might be a bit too complex and prone to errors. I am glad to see that some simplifications are already being proposed in issue #86.

It would be useful if this rule would be formally written up somewhere in the dictionary.

Also, with the clarification in place,  the ``_type.purpose`` of the following data items should be updated:
```
save_diffrn_scale_group.code
save_diffrn_standard_refln.code
save_geom_angle.atom_site_label_1
save_geom_angle.atom_site_label_3
save_geom_bond.atom_site_label_1
save_geom_bond.atom_site_label_2
save_geom_contact.atom_site_label_1
save_geom_contact.atom_site_label_2
save_geom_hbond.atom_site_label_A
save_geom_hbond.atom_site_label_D
save_geom_hbond.atom_site_label_H
save_geom_torsion.atom_site_label_1
save_geom_torsion.atom_site_label_2
save_geom_torsion.atom_site_label_3
save_geom_torsion.atom_site_label_4
save_valence_param.atom_1
save_valence_param.atom_2
save_valence_param.ref_id
save_atom_site.label
save_atom_site_aniso.label
save_atom_type_scat.symbol
```

**Comment by vaitkus** [comment]
This problem with data items ``_diffrn_scale_group.code`` and ``_diffrn_standard_refln.code`` is also addressed in issue #124.
